config/config-mp/src/main/java/io/helidon/config/mp/MpConfigImpl.java

```diff
@@ -313,35 +334,47 @@ private <T> T convert(String propertyName, Class<T> type, String value) {
         }
     }
 
-    private Optional<ConfigValue> findConfigValue(String propertyName) {
+    private Optional<ConfigValue> findConfigValue(String propertyName, Optional<String> profiledPropertyName) {
         for (ConfigSource source : sources) {
-            String value = source.getValue(propertyName);
+            String selectedProperty = null;
+            String value = null;
+            if (profiledPropertyName.isPresent()) {
+                // Try profiled property first
+                selectedProperty = profiledPropertyName.get();
+                value = source.getValue(profiledPropertyName.get());
+            }
+            if (value == null) {
+                selectedProperty = propertyName;
+                value = source.getValue(propertyName);
+            }
 
             if (null == value) {
                 // not in this one
                 continue;
             }
-
+            String rawValue = value;
+            String name = source.getName();
+            int ordinal = source.getOrdinal();
+            final String propName = selectedProperty;
             if (value.isEmpty()) {
                 if (LOGGER.isLoggable(Level.TRACE)) {
-                    LOGGER.log(Level.TRACE, "Found property " + propertyName
+                    LOGGER.log(Level.TRACE, "Found property " + propName
                                           + " in source " + source.getName()
                                           + " and it is empty (removed)");
                 }
-                return Optional.empty();
+                return Optional.of(new ConfigValueImpl(propName, null, rawValue, name, ordinal));
```

Comment:
Why is this not returning an empty optional? I am missing an explanation of this change (and once more further down).

Reply:
This is the point 1 I wrote at the beginning:
1. In one case we were returning an empty optional of the ConfigValue. It needs to have some properties, like ordinal and name.

Test fails otherwise, I don't recall the reason.

Reply:
There are three tests failing otherwise:
```
[ERROR] Failures: 
[ERROR]   CDIPropertyExpressionsTest>Arquillian.run:138->badExpansion:84 expected [test] but found [null]
[ERROR]   PropertyExpressionsTest>Arquillian.run:138->noExpressionButConfigValue:145 expected [test] but found [null]
[ERROR]   PropertyExpressionsTest>Arquillian.run:138->noExpressionComposedButConfigValue:171 expected [test] but found [null]
```

I am guided by the test results. The tests do not give any explication, so I am afraid I cannot tell you why it has to behave in this way.